build/build.go

```diff
@@ -703,11 +706,49 @@ func Invoke(ctx context.Context, cfg ContainerConfig) error {
 		}
 		defer ctr.Release(context.TODO())
 
+		imgData := res.Metadata[exptypes.ExporterImageConfigKey]
+		var img *specs.Image
+		if len(imgData) > 0 {
+			img = &specs.Image{}
+			if err := json.Unmarshal(imgData, img); err != nil {
+				fmt.Println(err)
+				return nil, err
+			}
+		}
+
+		user := ""
+		if cfg.User != nil {
+			user = *cfg.User
+		} else if img != nil {
+			user = img.Config.User
+		}
+		cwd := ""
+		if cfg.Cwd != nil {
+			cwd = *cfg.Cwd
+		} else if img != nil {
+			cwd = img.Config.WorkingDir
+		}
+		env := append([]string{}, cfg.Env...)
+		if img != nil {
+			env = append(env, img.Config.Env...)
+		}
```

Comment:
I think environment variables in `ContainerConfig` should be prioritized.
Current behaviour looks like envvars set by user aren't set if the image already has them.

e.g. `foo=bar` isn't set:

```dockerfile
FROM busybox
RUN echo hello > /hello
ENV foo=foo
```

```console
$ BUILDX_EXPERIMENTAL=1 /tmp/out/buildx build --invoke env=foo=bar,env=foo2=bar2 /tmp/ctx
[+] Building 4.1s (6/6) FINISHED                                                                                                   
 => [internal] load .dockerignore                                                                                             0.0s
 => => transferring context: 2B                                                                                               0.0s
 => [internal] load build definition from Dockerfile                                                                          0.0s
 => => transferring dockerfile: 86B                                                                                           0.0s
 => [internal] load metadata for docker.io/library/busybox:latest                                                             3.6s
 => [auth] library/busybox:pull token for registry-1.docker.io                                                                0.0s
 => [1/2] FROM docker.io/library/busybox@sha256:ad9bd57a3a57cc95515c537b89aaa69d83a6df54c4050fcf2b41ad367bec0cd5              0.3s
 => => resolve docker.io/library/busybox@sha256:ad9bd57a3a57cc95515c537b89aaa69d83a6df54c4050fcf2b41ad367bec0cd5              0.0s
 => => sha256:729ce43e2c915c3463b620f3fba201a4a641ca5a282387e233db799208342a08 772.99kB / 772.99kB                            0.2s
 => => extracting sha256:729ce43e2c915c3463b620f3fba201a4a641ca5a282387e233db799208342a08                                     0.0s
 => [2/2] RUN echo hello > /hello                                                                                             0.1s
Launching interactive container. Press Ctrl-a-c to switch to monitor console
/ # env
OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc
SHLVL=1
foo=foo
HOME=/root
foo2=bar2
TERM=xterm
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=unix:///dev/otel-grpc.sock
PWD=/
OTEL_TRACES_EXPORTER=otlp
```

Reply:
:tada: oops, good catch, will fixup that up :smile: